core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -865,9 +866,15 @@ public void forceMerge(final boolean flush, int maxNumSegments, boolean onlyExpu
                 store.decRef();
             }
         } catch (Throwable t) {
-            ForceMergeFailedEngineException ex = new ForceMergeFailedEngineException(shardId, t);
-            maybeFailEngine("force merge", ex);
-            throw ex;
+            maybeFailEngine("force merge", new ForceMergeFailedEngineException(shardId, t));
+            // wrap things we don't know in a ForceMergeFailedEngineException but let exceptions that indicate a closed shard bubble up
+            if (t instanceof AlreadyClosedException) {
```

Comment:
I think we should just rethrow `t` and remove that exception. We can't handle this just like refresh since a failing refresh means dataloss while this one might just be a configuration issue or so. We should just pass t to `maybeFailEngine` and rethrow IMO